### PR TITLE
chore: Add links and headers

### PR DESCRIPTION
**Pull request by rkoron007**
Fix links and headers and add redirect. [Jira ticket](https://hashicorp.atlassian.net/browse/IPE-606?atlOrigin=eyJpIjoiMTVlZTlkZmNlMjUyNDY5YWE1YjhkZjNjODAwYzNkMDEiLCJwIjoiaiJ9)

### Description

In plain English, describe your approach to addressing the issue linked above. For example, if you made a particular design decision, let us know why you chose this path instead of another solution.

### Checklist

- [x] I have updated the PR title to match [CDKTF's style guide](https://github.com/hashicorp/terraform-cdk/blob/main/CONTRIBUTING.md#pull-requests-1)
- [ ] I have run the linter on my code locally
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation if applicable
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works if applicable
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: website/docs/cdktf/concepts/remote-backends.mdx [hunk 1]
```diff
@@ -345,7 +345,7 @@ func main() {
 
 </CodeTabs>
 
-3. Run `cdktf diff <stack name> --migrate-state` to migrate the state into Terraform Cloud / Terraform Enterprise.
+3. Run `cdktf diff <stack name> --migrate-state` to migrate the state into HCP Terraform / Terraform Enterprise.
```
Comment (rkoron007) [inline website/docs/cdktf/concepts/remote-backends.mdx:348]:
```suggestion
3. Run `cdktf diff <stack name> --migrate-state` to migrate the state into HCP Terraform  or Terraform Enterprise.
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 2]
```diff
@@ -6,6 +6,6 @@ contact_links:
   - name: ❓ Question
     url: https://discuss.hashicorp.com/c/terraform-core/cdk-for-terraform/47
     about: 🙋 For usage questions that may not require a core maintainer to answer, post in our community forum
-  - name: Terraform Cloud/Enterprise Troubleshooting
+  - name: HCP Terraform/Enterprise Troubleshooting
```
Comment (ritsok) [inline .github/ISSUE_TEMPLATE/config.yml:9]:
```suggestion
  - name: HCP Terraform/Terraform Enterprise Troubleshooting
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 3]
```diff
@@ -6,6 +6,6 @@ contact_links:
...
     url: https://support.hashicorp.com/hc/en-us/requests/new
-    about: For issues related to the Terraform Cloud/Enterprise platform, please submit a HashiCorp support request or email tf-cloud@hashicorp.support
+    about: For issues related to the HCP Terraform/Enterprise platform, please submit a HashiCorp support request or email tf-cloud@hashicorp.support
```
Comment (ritsok) [inline .github/ISSUE_TEMPLATE/config.yml:11]:
```suggestion
    about: For issues related to the HCP Terraform/Terraform Enterprise platform, please submit a HashiCorp support request or email tf-cloud@hashicorp.support
```

File: website/docs/cdktf/create-and-deploy/hcp-terraform.mdx [hunk 4]
```diff
@@ -0,0 +1,185 @@
+---
+page_title: Connect to HCP Terraform / Enterprise - CDK for Terraform
+description: >-
+  Use HCP Terraform with CDKTF to store Terraform state remotely, manage secrets, apply policies, and more.
+---
+
+# Connect to HCP Terraform / Enterprise
```
Comment (ritsok) [inline website/docs/cdktf/create-and-deploy/hcp-terraform.mdx:7]:
```suggestion
# Connect to HCP Terraform / Terraform Enterprise
```

File: website/docs/cdktf/create-and-deploy/hcp-terraform.mdx [hunk 5]
```diff
@@ -0,0 +1,185 @@
+---
+page_title: Connect to HCP Terraform / Enterprise - CDK for Terraform
```
Comment (ritsok) [inline website/docs/cdktf/create-and-deploy/hcp-terraform.mdx:2]:
```suggestion
page_title: Connect to HCP Terraform / Terraform Enterprise - CDK for Terraform
```

File: website/docs/cdktf/create-and-deploy/hcp-terraform.mdx [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+description: >-
+  Use HCP Terraform with CDKTF to store Terraform state remotely, manage secrets, apply policies, and more.
+---
+
+# Connect to HCP Terraform / Enterprise
+
+This page explains how to connect HCP Terraform / Enterprise to your CDK for Terraform (CDKTF) application and the benefits of using these products together.
```
Comment (ritsok) [inline website/docs/cdktf/create-and-deploy/hcp-terraform.mdx:9]:
```suggestion
This page explains how to connect HCP Terraform / Terraform Enterprise to your CDK for Terraform (CDKTF) application and the benefits of using these products together.
```

File: website/docs/cdktf/create-and-deploy/hcp-terraform.mdx [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+
+Terraform Enterprise lets you set up a self-hosted distribution of HCP Terraform and is ideal for organizations with strict security and compliance requirements.
+
+## When to use HCP Terraform or Terraform Enterprise?
+
+Terraform uses persisted state data to keep track of the real-world resources it manages. By default, Terraform writes state to a local file. We recommend integrating with HCP Terraform to store state remotely. This prevents accidental loss of locally stored state and lets multiple people access the state data so they can work together on that collection of infrastructure resources.
+
+Other benefits of HCP Terraform and Terraform Enterprise include the ability to run Terraform remotely, [manage variables and secrets](#managing-variables-and-secrets), and enforce [policies for sets of infrastructure](#policy-enforcement).
+
+## Set up CDKTF with HCP Terraform / Enterprise
```
Comment (ritsok) [inline website/docs/cdktf/create-and-deploy/hcp-terraform.mdx:27]:
```suggestion
## Set up CDKTF with HCP Terraform / Terraform Enterprise
```

File: website/docs/cdktf/create-and-deploy/hcp-terraform.mdx [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+   app.synth();
+   ```
+
+3. Run `cdktf login` / `cdktf login --tfe-hostname=tfe.my-company.com` and log in to HCP Terraform / Enterprise.
+
+Your CDKTF application is connected to the HCP Terraform / Enterprise workspace you specified in the `RemoteBackend`. You must set the workspace to [Remote Execution mode](/terraform/cloud-docs/workspaces/settings#execution-mode) if you want to store state and run Terraform operations remotely.
+
+### VCS-driven Workflow
+
+Check your synthesized code into version control and connect your HCP Terraform / Enterprise workspace to that repository with the VCS-driven workflow. You can configure the workspace to trigger Terraform runs based on merges and commits to the repository. Refer to [Deployment Patterns](/terraform/cdktf/create-and-deploy/deployment-patterns#deploy-with-terraform-and-hcp-terraform) for more details about how to use CDKTF with the VCS-driven workflow.
```
Comment (ritsok) [inline website/docs/cdktf/create-and-deploy/hcp-terraform.mdx:136]:
```suggestion
Check your synthesized code into version control and connect your HCP Terraform / Terraform Enterprise workspace to that repository with the VCS-driven workflow. You can configure the workspace to trigger Terraform runs based on merges and commits to the repository. Refer to [Deployment Patterns](/terraform/cdktf/create-and-deploy/deployment-patterns#deploy-with-terraform-and-hcp-terraform) for more details about how to use CDKTF with the VCS-driven workflow.
```

File: website/docs/cdktf/create-and-deploy/hcp-terraform.mdx [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+
+## Managing Variables and Secrets
+
+HCP Terraform / Enterprise [variables](/terraform/cloud-docs/workspaces/variables) let you define the variables and secrets that Terraform uses during remote operations. You can set variables specifically for each workspace or you can create variable sets to reuse the same variables across multiple workspaces. For example, you could define a variable set of provider credentials and automatically apply it to all of the workspaces using that provider.
```
Comment (ritsok) [inline website/docs/cdktf/create-and-deploy/hcp-terraform.mdx:140]:
```suggestion
HCP Terraform / Terraform Enterprise [variables](/terraform/cloud-docs/workspaces/variables) let you define the variables and secrets that Terraform uses during remote operations. You can set variables specifically for each workspace or you can create variable sets to reuse the same variables across multiple workspaces. For example, you could define a variable set of provider credentials and automatically apply it to all of the workspaces using that provider.
```

File: website/docs/cdktf/create-and-deploy/hcp-terraform.mdx [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+
+### Variable Return Values
+
+The return value for `TerraformVariable` is an object with various representations of the value as attributes. You can pass this object as a string with `value.stringValue`, as a number with `value.numberValue`, or as a list with `value.listValue`.
+
+This means that there is no actual value in the variable field when CDKTF synthesizes your application. Instead, CDKTF uses a [Token](/terraform/cdktf/concepts/tokens), which represents a value that is unknown until Terraform applies your configuration. You cannot use tokens in dynamic checks during runtime. For example, `if (value.listValue.length > 42) {` always returns `false` because tokenized lists have a static length of one item.
+
+## Continuous Integration
+
+To run HCP Terraform / Enterprise in a CI workflow, you can either use [HCP Terraform / Enterprise's VCS-driven workflow](/terraform/cdktf/create-and-deploy/deployment-patterns#deploy-with-terraform-and-hcp-terraform) or use a [general-purpose CI to trigger the run in HCP Terraform / Enterprise](/terraform/cdktf/create-and-deploy/deployment-patterns#github-actions-ci-and-hcp-terraform).
```
Comment (ritsok) [inline website/docs/cdktf/create-and-deploy/hcp-terraform.mdx:177]:
```suggestion
To run HCP Terraform / Terraform Enterprise in a CI workflow, you can either use [HCP Terraform / Enterprise's VCS-driven workflow](/terraform/cdktf/create-and-deploy/deployment-patterns#deploy-with-terraform-and-hcp-terraform) or use a [general-purpose CI to trigger the run in HCP Terraform / Enterprise](/terraform/cdktf/create-and-deploy/deployment-patterns#github-actions-ci-and-hcp-terraform).
```

**Review by ritsok — APPROVED** [review]

File: website/docs/cdktf/cli-reference/cli-configuration.mdx [hunk 11]
```diff
@@ -30,7 +30,7 @@ Commands:
   cdktf destroy [OPTIONS] <stacks..>  Destroy the given stacks
   cdktf diff [stack] [OPTIONS]        Perform a diff (terraform plan) for the given stack                                                                                                         [aliases: plan]
   cdktf list [OPTIONS]                List stacks in app.
-  cdktf login                         Retrieves an API token to connect to Terraform Cloud.
+  cdktf login                         Retrieves an API token to connect to HCP Terraform.
```
Comment (BrianMMcClain) [inline website/docs/cdktf/cli-reference/cli-configuration.mdx:33]:
```suggestion
  cdktf login                         Retrieves an API token to connect to Terraform Cloud.
```

In the tutorial updates, we had decided to leave the CLI output unaltered. Do we want to do the same in the docs?

File: website/docs/cdktf/cli-reference/commands.mdx [hunk 12]
```diff
@@ -74,7 +74,7 @@ diff        -- Perform a diff (terraform plan) for the given stack
 get         -- Generate CDK Constructs for Terraform providers and modules.
 init        -- Create a new cdktf project from a template.
 list        -- List stacks in app.
-login       -- Retrieves an API token to connect to Terraform Cloud.
+login       -- Retrieves an API token to connect to HCP Terraform.
```
Comment (BrianMMcClain) [inline website/docs/cdktf/cli-reference/commands.mdx:77]:
```suggestion
login       -- Retrieves an API token to connect to Terraform Cloud.
```

**Review by BrianMMcClain — APPROVED** [review]

**Pull request merged**